src/main/java/org/candlepin/resource/ConsumerResource.java

```diff
@@ -1063,15 +1064,30 @@ public Consumer createConsumerFromDTO(ConsumerDTO consumer, ConsumerType type, P
         }
     }
 
-    private List<ActivationKey> findActivationKeys(
+    /**
+     * We need to make sure the requested keys are available. In addition, the order in
+     *  which they are specified in the call should be maintained for their later application.
+     *
+     * @param principal
+     * @param owner
+     * @param keyStrings
+     * @return
+     */
+    private List<ActivationKey> findActivationKeysInOrder(
         Principal principal, Owner owner, Set<String> keyStrings) {
```

Comment:
Because we're still accepting a set here, we've already lost the requested order. Given that we also do the parsing in this class, we probably should just pass in the original string here rather than a pre-parsed collection. That avoids having to re-wrap the items, and might clean up the comparator a bit.